src/kOS.Safe/Execution/CPU.cs

```diff
@@ -413,11 +415,13 @@ public void AssertValidDelegateCall(IUserDelegate userDelegate)
         {
             if (userDelegate.ProgContext != currentContext)
             {
-                throw new KOSInvalidDelegateContextException(
-                    (currentContext == contexts[0] ? "the interpreter" : "a program"),
-                    (currentContext == contexts[0] ? "a program" : "the interpreter")
-                    );
-            }
+                string currentContextName = (currentContext == contexts[0] ? "the interpreter" : "a program");
+                string delegateContextName = (
+                    userDelegate.ProgContext == contexts[0] ? currentContextName = "the interpreter" : (
+                        currentContext == contexts[0] ? "a program" : "a different program from a previous run" ) );
+
+                throw new KOSInvalidDelegateContextException(currentContextName, delegateContextName);
+           }
```

Comment:
Good catch.  It was an error from cutting and pasting as I arranged the statement around.  The ``currentContextName =`` should have been deleted.